src/block_service/block_service_manager.cpp

```diff
@@ -118,6 +118,13 @@ error_code block_service_manager::download_file(const std::string &remote_dir,
                                                 block_filesystem *fs,
                                                 /*out*/ uint64_t &download_file_size)
 {
+    // local file exists
+    const std::string local_file_name = utils::filesystem::path_combine(local_dir, file_name);
+    if (utils::filesystem::file_exists(local_file_name)) {
+        ddebug_f("local file({}) exists", local_file_name);
+        return ERR_DOWNLOADED;
```

Comment:
```suggestion
        return ERR_PATH_ALREADY_EXIST;
```

Using the existing error code is enough, and is more accurate here. Because you didn't check carefully if the file is really downloaded, as you didn't check the md5sum.